tasks/main.yml

```diff
@@ -61,6 +61,41 @@
   changed_when: false
   check_mode: no
 
+# Install the 2FA packages and setup the config in PAM and SSH
+
+- block:
+  - name: Install google authenticator PAM module
+    apt: name=libpam-google-authenticator state=installed
+    when: ansible_distribution == 'Debian' or ansible_distribution == 'Ubuntu'
+
+  - name: Install google authenticator PAM module
+    yum: name=google-authenticator state=installed
+    when: ansible_os_family == 'RedHat' or ansible_os_family == 'Oracle Linux'
+
+  - name: Add google auth module to PAM
+    lineinfile: "dest=/etc/pam.d/sshd state=present insertbefore=BOF regexp='pam_google_authenticator.so$' line='auth required pam_google_authenticator.so'"
+    notify: restart sshd
+
+  - name: Remove password auth from PAM
```

Comment:
Interesting. I will go and have a look at the module and see if it would work. I was trying to ensure I was not bringing in any additional external dependencies but using a module for this would be nicer.